app/src/js/components/Table/Table.js

```diff
@@ -8,224 +8,209 @@ import omitBy from 'lodash/omitBy';
-    this.queryNewSort = this.queryNewSort.bind(this);
-    this.updateSelection = this.updateSelection.bind(this);
-    this.onBulkActionSuccess = this.onBulkActionSuccess.bind(this);
-    this.onBulkActionError = this.onBulkActionError.bind(this);
-    this.getQueryConfig = this.getQueryConfig.bind(this);
-
-    const initialPage = 1;
-    const initialSortId = props.sortId;
-    const initialOrder = 'desc';
-    const sortProps = initialSortId ? [{ id: initialSortId, desc: true }] : [];
-
-    this.state = {
-      page: initialPage,
-      sortId: initialSortId,
-      order: initialOrder,
-      sortProps,
-      selected: [],
-      clearSelected: false,
-      infix: null,
-      queryConfig: {
-        page: initialPage,
-        sort_key: buildSortKey(sortProps),
-        ...(props.query || {}),
-      },
-      params: {},
-      completedBulkActions: 0,
-      bulkActionError: null,
-    };
-  }
-
-  componentDidUpdate(prevProps) {
-    const { list, query, queryParams } = this.props;
-
-    if (!isEqual(query, prevProps.query)) {
-      // eslint-disable-next-line react/no-did-update-set-state
-      this.setState({ queryConfig: this.getQueryConfig({}, query) });
-    }
-
+const List = ({
+  action,
+  bulkActions,
+  children,
+  data,
+  dispatch,
+  filterAction,
+  filterClear,
+  initialSortId,
+  list,
+  onSelect,
+  query,
+  queryParams,
+  rowId,
+  tableColumns,
+}) => {
+  const { data: listData, error: listError, inflight: listInflight, meta } = list;
+  const { count, limit } = meta;
+  const tableData = data || listData;
+
+  const [selected, setSelected] = useState([]);
+  const [clearSelected, setClearSelected] = useState(false);
+
+  const [queryConfig, setQueryConfig] = useState({
+    page: 1,
+    sort_key: buildSortKey(initialSortId ? [{ id: initialSortId, desc: true }] : []),
+    ...(query || {}),
+  });
+  const [params, setParams] = useState({});
+  const [bulkActionMeta, setBulkActionMeta] = useState({
+    completedBulkActions: 0,
+    bulkActionError: null,
+  });
+
+  const { page } = queryConfig;
+  const { bulkActionError, completedBulkActions } = bulkActionMeta;
+  const { limit: limitQueryParam, page: pageQueryParam, ...queryFilters } = queryParams;
+
+  const hasActions = Array.isArray(bulkActions) && bulkActions.length > 0;
+
+  useEffect(() => {
+    setQueryConfig((prevQueryConfig) => ({
+      ...prevQueryConfig,
+      ...getQueryConfig({})
+    }));
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [JSON.stringify(query)]);
+
+  useEffect(() => {
     // Remove parameters with null or undefined values
-    const params = omitBy(list.params, isNil);
+    const newParams = omitBy(list.params, isNil);
 
-    if (!isEqual(params, this.state.params)) {
-      // eslint-disable-next-line react/no-did-update-set-state
-      this.setState({ params }, () => this.setState({
-        queryConfig: this.getQueryConfig(),
+    if (!isEqual(newParams, params)) {
```

Comment:
do we need this test? wont the useEffect dependencies handle that?   Or is this special because some params can become Null/undefined?

Reply:
Yeah exactly, the useEffect will be triggered if values in list.params are null, but we only want to update the params if the non-null values change. Truthfully, not 100% sure if keeping track of the params like this is necessary, but was afraid to remove it, so wrote it in a way that would mirror the previous functionality.